examples/GitHubLabeler/GitHubLabeler/Predictor.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.ML;
+using Microsoft.ML.Trainers;
+using Microsoft.ML.Transforms;
+
+namespace GitHubLabeler
+{
+    internal class Predictor
+    {
+        private static string AppPath => Path.GetDirectoryName(Environment.GetCommandLineArgs()[0]);
+
+        private static string DataPath => Path.Combine(AppPath, "Data", "corefx_issues.tsv");
+
+        private static string ModelPath => Path.Combine(AppPath, "Models", "Model.zip");
+
+        private static PredictionModel<GitHubIssue, GitHubIssuePrediction> _model;
+
+        public static async Task TrainAsync()
+        {
+            var pipeline = new LearningPipeline();
+
+            pipeline.Add(new TextLoader<GitHubIssue>(DataPath, useHeader: true));
```

Comment:
>   pipeline.Add(new TextLoader<GitHubIssue>(DataPath, useHeader: true)); [](start = 9, length = 72)

How is this line even compiling? Are you using some outdated nuget package? The loader API has been changed.

Reply:
These are samples, which means they are building against the latest package that is published to nuget.org, not what is currently in master. Hence, they are working as we're consuming what our customers would, which is what we want.

Reply:
Hmmm. The alternate argument is, if they're checked in with the repo, they ought to target the code *in the repo*, since then the samples are simply always up to date, if we make them part of our build/test cycle. Indeed until I saw that we were using this outdated stuff, I had fully expected that to be the case.

The nuget story makes the samples far more difficult to maintain as a code artifact. Especially in these days, when the API is still in heavy flux. I'm considering two worlds:

* The samples are incrementally changed so that as we move towards 0.2, changing the samples *one at a time* by the person authoring that PR (and so, hopefully, understands how it should be use), I see how development happens. Also at any given point, **the samples are correct**.
* The samples remain unchanged. Come release time, it is *someone's* job (whose?) to comprehend all PRs against the API in that time, so they have a clue of how the sample should be updated, and then we hope they do all that correctly. Meanwhile the release is blocked on this point.

Generally anything that reduces maintainability and adds costs to our release process, is something we ought to be hostile towards.

---
In reply to: [190764164](https://github.com/dotnet/machinelearning/pull/238#discussion_r190764164) [](ancestors = 190764164)

Reply:
Actually yeah, I was thinking about it, and the nuget approach just seems deeply wrong. You couldn't even write the PR to update the samples until *after* you've actually published the nuget. This means that any release branch we make will either have *by design* outdated samples, or we will have to accept a world in which the nuget published for version X never corresponds to the branch `release/X`. Regarding it being what we want, I'm not sure who wants that, but whoever they are, let's have them want something else. :)

---
In reply to: [190949466](https://github.com/dotnet/machinelearning/pull/238#discussion_r190949466) [](ancestors = 190949466,190764164)